sched/semaphore/spinlock.c

```diff
@@ -446,4 +446,225 @@ void spin_clrbit(FAR volatile cpu_set_t *set, unsigned int cpu,
 }
 #endif
 
+#ifdef CONFIG_RW_SPINLOCK
+
+/****************************************************************************
+ * Name: read_lock
+ *
+ * Description:
+ *   If this task does not already hold the spinlock, then loop until the
+ *   spinlock is successfully locked.
+ *
+ *   This implementation is non-reentrant and set a bit of lock.
+ *
+ *  The priority of reader is higher than writter if a reader hold the
+ *  lock, a new reader can get its lock but writer can't get this lock.
+ *
+ * Input Parameters:
+ *   lock - A reference to the spinlock object to lock.
+ *
+ * Returned Value:
+ *   None.  When the function returns, the spinlock was successfully locked
+ *   by this CPU.
+ *
+ * Assumptions:
+ *   Not running at the interrupt level.
+ *
+ ****************************************************************************/
+
+void read_lock(FAR volatile rwlock_t *lock)
+{
+  rwlock_t old;
+
+  while (true)
+    {
+      old = atomic_load(lock);
+
+      if (old >= RW_SP_UNLOCKED &&
```

Comment:
```
if (old < 0)
  {
    SP_DSB();
    SP_WFE();
  }
else if (atomic_compare_exchange_strong(lock, &old, old + 1))
  {
    break;
  }
```

Reply:
I want to modify 
old < 0 ---> old <= RW_SP_WRITE_LOCKED
It's more readable.

Reply:
Yes, please go ahead.